aten/src/ATen/native/Math.h

```diff
@@ -292,7 +292,14 @@ static inline double calc_digamma(double x) {
       // If the argument is a negative integer, NaN is returned
       return std::numeric_limits<double>::quiet_NaN();
     }
-    return calc_digamma(1 - x) - c10::pi<double> / tan(c10::pi<double> * x);
+    // In the following piece of code we extract fractiontional part of x (radian)
+    // in order to multiply by pi. Tangent function has period pi (irrational number),
+    // which is slighly different than numerically estimated pi provided here. When x
+    // is big enough multiplying it with estimated pi would lead to much bigger difference.
+    // This difference can lead to bir errors in extreme values.
```

Comment:
"fractiontional" -> "fractional"
"bir errors" -> "big errors"

But this comment can also be simplified a little:

```
Extracts the fractional part of x as r, since tan(pi * r) is more numerically accurate than tan(pi * x). While these operations are mathematically equivalent since both x and r are in radians and tan() has a periodicity of pi, in practice the computation of pi * x is a source of error (when |x| > 1). 
```

Reply:
this is indeed very nice choice of explanation what happens in the following code. Thanks for the suggestion.